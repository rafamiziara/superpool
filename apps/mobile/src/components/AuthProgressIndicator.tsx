import { observer } from 'mobx-react-lite'
import React from 'react'
import { Text, View } from 'react-native'
import { useAuthenticationStore } from '../stores'
import { ProgressIndicator } from './ProgressIndicator'

interface AuthProgressIndicatorProps {
  className?: string
}

export const AuthProgressIndicator = observer(function AuthProgressIndicator({ className = '' }: AuthProgressIndicatorProps) {
  const authStore = useAuthenticationStore()

  const steps = authStore.getAllSteps()
  const currentStepIndex = steps.findIndex((step) => step.step === authStore.currentStep)

  if (steps.length === 0) {
    return null
  }

  return (
    <View className={`space-y-4 ${className}`} testID="auth-progress-indicator">
      <ProgressIndicator totalSteps={steps.length} currentStep={currentStepIndex >= 0 ? currentStepIndex : -1} />

      <View className="space-y-2">
        {authStore.currentStep && (
          <Text className="text-center text-sm font-medium text-foreground" testID="current-step-title">
            {authStore.getStepInfo(authStore.currentStep)?.title}
          </Text>
        )}

        {authStore.currentStep && (
          <Text className="text-center text-xs text-muted-foreground" testID="current-step-description">
            {authStore.getStepInfo(authStore.currentStep)?.description}
          </Text>
        )}

        {authStore.progressError && (
          <Text className="text-center text-xs text-destructive" testID="progress-error">
            {authStore.progressError}
          </Text>
        )}
      </View>

      <View className="flex-row justify-center space-x-4">
        <Text className="text-xs text-muted-foreground" testID="progress-stats">
          {authStore.completedSteps.size} of {steps.length} completed
        </Text>

        {authStore.isProgressComplete && (
          <Text className="text-xs text-green-600 font-medium" testID="progress-complete">
            âœ“ Complete
          </Text>
        )}
      </View>
    </View>
  )
})
