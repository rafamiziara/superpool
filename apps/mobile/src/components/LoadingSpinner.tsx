import React from 'react';
import { ActivityIndicator, View } from 'react-native';

interface LoadingSpinnerProps {
  size?: 'small' | 'large';
  className?: string;
  color?: string;
}

export function LoadingSpinner({ 
  size = 'large', 
  className = '', 
  color = '#2563eb' // Default to primary color
}: LoadingSpinnerProps) {
  return (
    <View className={`items-center justify-center ${className}`}>
      <ActivityIndicator size={size} color={color} />
    </View>
  );
}