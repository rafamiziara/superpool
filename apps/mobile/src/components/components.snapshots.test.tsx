import React from 'react'
import { mockStorePresets, render, renderWithStore } from '../test-utils'
import { AuthProgressIndicator } from './AuthProgressIndicator'
import { LoadingSpinner } from './LoadingSpinner'
import { ProgressIndicator } from './ProgressIndicator'

describe('Component Snapshots', () => {
  describe('LoadingSpinner Snapshots', () => {
    it('should match snapshot with default props', () => {
      const component = render(<LoadingSpinner />)
      expect(component.toJSON()).toMatchSnapshot('LoadingSpinner-default')
    })

    it('should match snapshot with small size', () => {
      const component = render(<LoadingSpinner size="small" />)
      expect(component.toJSON()).toMatchSnapshot('LoadingSpinner-small')
    })

    it('should match snapshot with custom color and className', () => {
      const component = render(<LoadingSpinner size="large" color="#ff6b6b" className="p-8 bg-gray-50" />)
      expect(component.toJSON()).toMatchSnapshot('LoadingSpinner-custom')
    })
  })

  describe('ProgressIndicator Snapshots', () => {
    it('should match snapshot with basic progress', () => {
      const component = render(<ProgressIndicator totalSteps={5} currentStep={2} />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-basic')
    })

    it('should match snapshot with first step active', () => {
      const component = render(<ProgressIndicator totalSteps={4} currentStep={0} />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-first-step')
    })

    it('should match snapshot with last step active', () => {
      const component = render(<ProgressIndicator totalSteps={3} currentStep={2} />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-last-step')
    })

    it('should match snapshot with single step', () => {
      const component = render(<ProgressIndicator totalSteps={1} currentStep={0} />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-single-step')
    })

    it('should match snapshot with custom className', () => {
      const component = render(<ProgressIndicator totalSteps={4} currentStep={1} className="p-4 m-2 bg-white rounded-lg" />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-custom-className')
    })

    it('should match snapshot with out-of-range currentStep', () => {
      const component = render(<ProgressIndicator totalSteps={3} currentStep={5} />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-out-of-range')
    })
  })

  describe('AuthProgressIndicator Snapshots', () => {
    it('should match snapshot with default state', async () => {
      const component = renderWithStore(<AuthProgressIndicator />)
      expect(component.toJSON()).toMatchSnapshot('AuthProgressIndicator-default')
    })

    it('should match snapshot with active authentication step', async () => {
      const mockStore = mockStorePresets.authenticating()
      mockStore.authenticationStore.startStep('generate-message')

      const component = renderWithStore(<AuthProgressIndicator />, { store: mockStore })
      expect(component.toJSON()).toMatchSnapshot('AuthProgressIndicator-active-step')
    })

    it('should match snapshot with completed authentication', async () => {
      const mockStore = mockStorePresets.authenticating()
      mockStore.authenticationStore.completeStep('firebase-auth') // Triggers completion

      const component = renderWithStore(<AuthProgressIndicator />, { store: mockStore })
      expect(component.toJSON()).toMatchSnapshot('AuthProgressIndicator-completed')
    })

    it('should match snapshot with error state', async () => {
      const mockStore = mockStorePresets.authenticationError()
      mockStore.authenticationStore.startStep('verify-signature')
      mockStore.authenticationStore.failStep('verify-signature', 'Network connection failed')

      const component = renderWithStore(<AuthProgressIndicator />, { store: mockStore })
      expect(component.toJSON()).toMatchSnapshot('AuthProgressIndicator-error-state')
    })

    it('should match snapshot with custom className', async () => {
      const mockStore = mockStorePresets.authenticating()
      mockStore.authenticationStore.startStep('request-signature')

      const component = renderWithStore(<AuthProgressIndicator className="p-6 bg-blue-50 rounded-xl" />, { store: mockStore })
      expect(component.toJSON()).toMatchSnapshot('AuthProgressIndicator-custom-className')
    })

    it('should match snapshot with multiple completed steps', async () => {
      const mockStore = mockStorePresets.authenticating()
      mockStore.authenticationStore.completeStep('generate-message')
      mockStore.authenticationStore.completeStep('request-signature')
      mockStore.authenticationStore.startStep('verify-signature')

      const component = renderWithStore(<AuthProgressIndicator />, { store: mockStore })
      expect(component.toJSON()).toMatchSnapshot('AuthProgressIndicator-multiple-completed')
    })
  })

  describe('Component Integration Snapshots', () => {
    it('should match snapshot of components used together', () => {
      const component = render(
        <>
          <LoadingSpinner size="small" className="mb-4" />
          <ProgressIndicator totalSteps={4} currentStep={1} className="mb-6" />
        </>
      )
      expect(component.toJSON()).toMatchSnapshot('Components-integration')
    })

    it('should match snapshot with various component states', () => {
      const component = render(
        <>
          <LoadingSpinner size="large" color="#2563eb" />
          <ProgressIndicator totalSteps={6} currentStep={3} />
          <LoadingSpinner size="small" color="#10b981" />
        </>
      )
      expect(component.toJSON()).toMatchSnapshot('Components-various-states')
    })
  })

  describe('Responsive Design Snapshots', () => {
    it('should match snapshot with different screen sizes simulation', () => {
      const component = render(<ProgressIndicator totalSteps={8} currentStep={4} className="w-full max-w-sm mx-auto" />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-responsive')
    })

    it('should match snapshot with loading spinner in different containers', () => {
      const component = render(
        <>
          <LoadingSpinner className="w-10 h-10" />
          <LoadingSpinner className="w-16 h-16" />
          <LoadingSpinner className="w-24 h-24" />
        </>
      )
      expect(component.toJSON()).toMatchSnapshot('LoadingSpinner-different-sizes')
    })
  })

  describe('Accessibility Snapshots', () => {
    it('should match snapshot with accessibility labels', () => {
      const component = render(<ProgressIndicator totalSteps={5} currentStep={2} className="accessible-progress" />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-accessibility')
    })
  })

  describe('Edge Cases Snapshots', () => {
    it('should match snapshot with zero steps', () => {
      const component = render(<ProgressIndicator totalSteps={0} currentStep={0} />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-zero-steps')
    })

    it('should match snapshot with negative current step', () => {
      const component = render(<ProgressIndicator totalSteps={3} currentStep={-1} />)
      expect(component.toJSON()).toMatchSnapshot('ProgressIndicator-negative-step')
    })

    it('should match snapshot with empty store state', async () => {
      const mockStore = mockStorePresets.unauthenticated()
      // Mock empty steps
      mockStore.authenticationStore.getAllSteps = () => []

      const component = renderWithStore(<AuthProgressIndicator />, { store: mockStore })
      expect(component.toJSON()).toMatchSnapshot('AuthProgressIndicator-empty-state')
    })
  })
})
