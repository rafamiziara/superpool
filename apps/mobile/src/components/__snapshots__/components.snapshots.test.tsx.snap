// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Snapshots Accessibility Snapshots should match snapshot with accessibility labels: ProgressIndicator-accessibility 1`] = `
<View
  className="flex-row justify-center items-center gap-2 accessible-progress"
  testID="progress-indicator-container"
>
  <View
    accessibilityLabel="Step 1 of 5 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-0"
  />
  <View
    accessibilityLabel="Step 2 of 5 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-1"
  />
  <View
    accessibilityLabel="Step 3 of 5 (current)"
    className="w-2 h-2 rounded-full bg-primary"
    testID="progress-step-2"
  />
  <View
    accessibilityLabel="Step 4 of 5 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-3"
  />
  <View
    accessibilityLabel="Step 5 of 5 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-4"
  />
</View>
`;

exports[`Component Snapshots AuthProgressIndicator Snapshots should match snapshot with active authentication step: AuthProgressIndicator-active-step 1`] = `
<View
  className="space-y-4 "
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (current)"
      className="w-2 h-2 rounded-full bg-primary"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  >
    <Text
      className="text-center text-sm font-medium text-foreground"
      testID="current-step-title"
    >
      Generate Auth Message
    </Text>
    <Text
      className="text-center text-xs text-muted-foreground"
      testID="current-step-description"
    >
      Creating authentication challenge
    </Text>
  </View>
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      1
       of 
      6
       completed
    </Text>
  </View>
</View>
`;

exports[`Component Snapshots AuthProgressIndicator Snapshots should match snapshot with completed authentication: AuthProgressIndicator-completed 1`] = `
<View
  className="space-y-4 "
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  />
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      2
       of 
      6
       completed
    </Text>
    <Text
      className="text-xs text-green-600 font-medium"
      testID="progress-complete"
    >
      âœ“ Complete
    </Text>
  </View>
</View>
`;

exports[`Component Snapshots AuthProgressIndicator Snapshots should match snapshot with custom className: AuthProgressIndicator-custom-className 1`] = `
<View
  className="space-y-4 p-6 bg-blue-50 rounded-xl"
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (current)"
      className="w-2 h-2 rounded-full bg-primary"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  >
    <Text
      className="text-center text-sm font-medium text-foreground"
      testID="current-step-title"
    >
      Request Signature
    </Text>
    <Text
      className="text-center text-xs text-muted-foreground"
      testID="current-step-description"
    >
      Sign message in your wallet app
    </Text>
  </View>
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      1
       of 
      6
       completed
    </Text>
  </View>
</View>
`;

exports[`Component Snapshots AuthProgressIndicator Snapshots should match snapshot with default state: AuthProgressIndicator-default 1`] = `
<View
  className="space-y-4 "
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  />
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      1
       of 
      6
       completed
    </Text>
  </View>
</View>
`;

exports[`Component Snapshots AuthProgressIndicator Snapshots should match snapshot with error state: AuthProgressIndicator-error-state 1`] = `
<View
  className="space-y-4 "
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  >
    <Text
      className="text-center text-xs text-destructive"
      testID="progress-error"
    >
      Network connection failed
    </Text>
  </View>
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      1
       of 
      6
       completed
    </Text>
  </View>
</View>
`;

exports[`Component Snapshots AuthProgressIndicator Snapshots should match snapshot with multiple completed steps: AuthProgressIndicator-multiple-completed 1`] = `
<View
  className="space-y-4 "
  testID="auth-progress-indicator"
>
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (current)"
      className="w-2 h-2 rounded-full bg-primary"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>
  <View
    className="space-y-2"
  >
    <Text
      className="text-center text-sm font-medium text-foreground"
      testID="current-step-title"
    >
      Verify Signature
    </Text>
    <Text
      className="text-center text-xs text-muted-foreground"
      testID="current-step-description"
    >
      Validating your signature
    </Text>
  </View>
  <View
    className="flex-row justify-center space-x-4"
  >
    <Text
      className="text-xs text-muted-foreground"
      testID="progress-stats"
    >
      3
       of 
      6
       completed
    </Text>
  </View>
</View>
`;

exports[`Component Snapshots Component Integration Snapshots should match snapshot of components used together: Components-integration 1`] = `
[
  <View
    className="items-center justify-center mb-4"
    testID="loading-spinner-container"
  >
    <ActivityIndicator
      color="#2563eb"
      size="small"
      testID="activity-indicator"
    />
  </View>,
  <View
    className="flex-row justify-center items-center gap-2 mb-6"
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 4 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 4 (current)"
      className="w-2 h-2 rounded-full bg-primary"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 4 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 4 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-3"
    />
  </View>,
]
`;

exports[`Component Snapshots Component Integration Snapshots should match snapshot with various component states: Components-various-states 1`] = `
[
  <View
    className="items-center justify-center "
    testID="loading-spinner-container"
  >
    <ActivityIndicator
      color="#2563eb"
      size="large"
      testID="activity-indicator"
    />
  </View>,
  <View
    className="flex-row justify-center items-center gap-2 "
    testID="progress-indicator-container"
  >
    <View
      accessibilityLabel="Step 1 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-0"
    />
    <View
      accessibilityLabel="Step 2 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-1"
    />
    <View
      accessibilityLabel="Step 3 of 6 (completed)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-2"
    />
    <View
      accessibilityLabel="Step 4 of 6 (current)"
      className="w-2 h-2 rounded-full bg-primary"
      testID="progress-step-3"
    />
    <View
      accessibilityLabel="Step 5 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-4"
    />
    <View
      accessibilityLabel="Step 6 of 6 (pending)"
      className="w-2 h-2 rounded-full bg-muted-foreground/30"
      testID="progress-step-5"
    />
  </View>,
  <View
    className="items-center justify-center "
    testID="loading-spinner-container"
  >
    <ActivityIndicator
      color="#10b981"
      size="small"
      testID="activity-indicator"
    />
  </View>,
]
`;

exports[`Component Snapshots Edge Cases Snapshots should match snapshot with empty store state: AuthProgressIndicator-empty-state 1`] = `null`;

exports[`Component Snapshots Edge Cases Snapshots should match snapshot with negative current step: ProgressIndicator-negative-step 1`] = `
<View
  className="flex-row justify-center items-center gap-2 "
  testID="progress-indicator-container"
>
  <View
    accessibilityLabel="Step 1 of 3 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-0"
  />
  <View
    accessibilityLabel="Step 2 of 3 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-1"
  />
  <View
    accessibilityLabel="Step 3 of 3 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-2"
  />
</View>
`;

exports[`Component Snapshots Edge Cases Snapshots should match snapshot with zero steps: ProgressIndicator-zero-steps 1`] = `
<View
  className="flex-row justify-center items-center gap-2 "
  testID="progress-indicator-container"
/>
`;

exports[`Component Snapshots LoadingSpinner Snapshots should match snapshot with custom color and className: LoadingSpinner-custom 1`] = `
<View
  className="items-center justify-center p-8 bg-gray-50"
  testID="loading-spinner-container"
>
  <ActivityIndicator
    color="#ff6b6b"
    size="large"
    testID="activity-indicator"
  />
</View>
`;

exports[`Component Snapshots LoadingSpinner Snapshots should match snapshot with default props: LoadingSpinner-default 1`] = `
<View
  className="items-center justify-center "
  testID="loading-spinner-container"
>
  <ActivityIndicator
    color="#2563eb"
    size="large"
    testID="activity-indicator"
  />
</View>
`;

exports[`Component Snapshots LoadingSpinner Snapshots should match snapshot with small size: LoadingSpinner-small 1`] = `
<View
  className="items-center justify-center "
  testID="loading-spinner-container"
>
  <ActivityIndicator
    color="#2563eb"
    size="small"
    testID="activity-indicator"
  />
</View>
`;

exports[`Component Snapshots ProgressIndicator Snapshots should match snapshot with basic progress: ProgressIndicator-basic 1`] = `
<View
  className="flex-row justify-center items-center gap-2 "
  testID="progress-indicator-container"
>
  <View
    accessibilityLabel="Step 1 of 5 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-0"
  />
  <View
    accessibilityLabel="Step 2 of 5 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-1"
  />
  <View
    accessibilityLabel="Step 3 of 5 (current)"
    className="w-2 h-2 rounded-full bg-primary"
    testID="progress-step-2"
  />
  <View
    accessibilityLabel="Step 4 of 5 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-3"
  />
  <View
    accessibilityLabel="Step 5 of 5 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-4"
  />
</View>
`;

exports[`Component Snapshots ProgressIndicator Snapshots should match snapshot with custom className: ProgressIndicator-custom-className 1`] = `
<View
  className="flex-row justify-center items-center gap-2 p-4 m-2 bg-white rounded-lg"
  testID="progress-indicator-container"
>
  <View
    accessibilityLabel="Step 1 of 4 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-0"
  />
  <View
    accessibilityLabel="Step 2 of 4 (current)"
    className="w-2 h-2 rounded-full bg-primary"
    testID="progress-step-1"
  />
  <View
    accessibilityLabel="Step 3 of 4 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-2"
  />
  <View
    accessibilityLabel="Step 4 of 4 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-3"
  />
</View>
`;

exports[`Component Snapshots ProgressIndicator Snapshots should match snapshot with first step active: ProgressIndicator-first-step 1`] = `
<View
  className="flex-row justify-center items-center gap-2 "
  testID="progress-indicator-container"
>
  <View
    accessibilityLabel="Step 1 of 4 (current)"
    className="w-2 h-2 rounded-full bg-primary"
    testID="progress-step-0"
  />
  <View
    accessibilityLabel="Step 2 of 4 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-1"
  />
  <View
    accessibilityLabel="Step 3 of 4 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-2"
  />
  <View
    accessibilityLabel="Step 4 of 4 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-3"
  />
</View>
`;

exports[`Component Snapshots ProgressIndicator Snapshots should match snapshot with last step active: ProgressIndicator-last-step 1`] = `
<View
  className="flex-row justify-center items-center gap-2 "
  testID="progress-indicator-container"
>
  <View
    accessibilityLabel="Step 1 of 3 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-0"
  />
  <View
    accessibilityLabel="Step 2 of 3 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-1"
  />
  <View
    accessibilityLabel="Step 3 of 3 (current)"
    className="w-2 h-2 rounded-full bg-primary"
    testID="progress-step-2"
  />
</View>
`;

exports[`Component Snapshots ProgressIndicator Snapshots should match snapshot with out-of-range currentStep: ProgressIndicator-out-of-range 1`] = `
<View
  className="flex-row justify-center items-center gap-2 "
  testID="progress-indicator-container"
>
  <View
    accessibilityLabel="Step 1 of 3 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-0"
  />
  <View
    accessibilityLabel="Step 2 of 3 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-1"
  />
  <View
    accessibilityLabel="Step 3 of 3 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-2"
  />
</View>
`;

exports[`Component Snapshots ProgressIndicator Snapshots should match snapshot with single step: ProgressIndicator-single-step 1`] = `
<View
  className="flex-row justify-center items-center gap-2 "
  testID="progress-indicator-container"
>
  <View
    accessibilityLabel="Step 1 of 1 (current)"
    className="w-2 h-2 rounded-full bg-primary"
    testID="progress-step-0"
  />
</View>
`;

exports[`Component Snapshots Responsive Design Snapshots should match snapshot with different screen sizes simulation: ProgressIndicator-responsive 1`] = `
<View
  className="flex-row justify-center items-center gap-2 w-full max-w-sm mx-auto"
  testID="progress-indicator-container"
>
  <View
    accessibilityLabel="Step 1 of 8 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-0"
  />
  <View
    accessibilityLabel="Step 2 of 8 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-1"
  />
  <View
    accessibilityLabel="Step 3 of 8 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-2"
  />
  <View
    accessibilityLabel="Step 4 of 8 (completed)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-3"
  />
  <View
    accessibilityLabel="Step 5 of 8 (current)"
    className="w-2 h-2 rounded-full bg-primary"
    testID="progress-step-4"
  />
  <View
    accessibilityLabel="Step 6 of 8 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-5"
  />
  <View
    accessibilityLabel="Step 7 of 8 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-6"
  />
  <View
    accessibilityLabel="Step 8 of 8 (pending)"
    className="w-2 h-2 rounded-full bg-muted-foreground/30"
    testID="progress-step-7"
  />
</View>
`;

exports[`Component Snapshots Responsive Design Snapshots should match snapshot with loading spinner in different containers: LoadingSpinner-different-sizes 1`] = `
[
  <View
    className="items-center justify-center w-10 h-10"
    testID="loading-spinner-container"
  >
    <ActivityIndicator
      color="#2563eb"
      size="large"
      testID="activity-indicator"
    />
  </View>,
  <View
    className="items-center justify-center w-16 h-16"
    testID="loading-spinner-container"
  >
    <ActivityIndicator
      color="#2563eb"
      size="large"
      testID="activity-indicator"
    />
  </View>,
  <View
    className="items-center justify-center w-24 h-24"
    testID="loading-spinner-container"
  >
    <ActivityIndicator
      color="#2563eb"
      size="large"
      testID="activity-indicator"
    />
  </View>,
]
`;
